### PR TITLE
Fix #316896: Banjo 5th string tablature problem

### PR DESCRIPTION
**Pull request by laturetab**
The banjo 5th string starts at the 5th fret, so valid fret
numbers are 0, 6, 7, 8, ... Musescore currently uses
0, 1, 2, 3...

This fix will make Musescore use the 0678 format for
banjo 5th string fret numbers. GuitarPro import will
now adjust from 0123 to 0678 format on the way in.

This fix does not change the UI or any file formats.

Resolves: *(direct link to the issue)*

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [ ] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Needs to based on 3.x, not 3.6.1. Or master

File: importexport/guitarpro/importgtp-gp5.cpp [hunk 1]
```diff
@@ -1278,7 +1278,7 @@ bool GuitarPro5::readNoteEffects(Note* note)
 			      case 9: fret += 28; break;
 			      default: fret += octave * 12;
 			      }
-		      harmonicNote->setString(note->string());
+		    harmonicNote->setString(note->string());
```
Comment (Jojo-Schmitz) [inline importexport/guitarpro/importgtp-gp5.cpp:1281]: Looks like whitespace damage (and yes, this file doesn't really follow the code style of MuseScore)

File: importexport/guitarpro/importgtp-gp4.cpp [hunk 2]
```diff
@@ -434,6 +434,8 @@ bool GuitarPro4::readNote(int string, int staffIdx, Note* note)
       // dead note represented as high numbers - fix to zero
       if (fretNumber > 99 || fretNumber == -1)
             fretNumber = 0;
+      if (fretNumber > 0 && string == 4)
+          fretNumber = staff->part()->instrument()->stringData()->adjustBanjo5thFret(fretNumber);
```
Comment (Jojo-Schmitz) [inline importexport/guitarpro/importgtp-gp4.cpp:438]: How does this work on any plucked string instrument with more than 4 strings that is not a Banjo?

File: importexport/guitarpro/importgtp-gp5.cpp [hunk 3]
```diff
@@ -1437,6 +1437,8 @@ bool GuitarPro5::readNote(int string, Note* note)
             note->setHeadGroup(NoteHead::Group::HEAD_CROSS);
             note->setGhost(true);
             }
+      if (fretNumber > 0 && string == 4)
+          fretNumber = staff->part()->instrument()->stringData()->adjustBanjo5thFret(fretNumber);
```
Comment (Jojo-Schmitz) [inline importexport/guitarpro/importgtp-gp5.cpp:1441]: How does this work on any plucked string instrument with more than 4 strings that is not a Banjo?

File: importexport/guitarpro/importgtp-gp6.cpp [hunk 4]
```diff
@@ -1224,6 +1224,8 @@ Fraction GuitarPro6::readBeats(QString beats, GPPartInfo* partInfo, Measure* mea
                                                 Staff* staff        = note->staff();
                                                 int fretNumber      = fretNum.toInt();
                                                 int musescoreString = staff->part()->instrument()->stringData()->strings() - 1 - stringNum.toInt();
+                                                if (fretNumber > 0 && musescoreString == 4)
+                                                      fretNumber = staff->part()->instrument()->stringData()->adjustBanjo5thFret(fretNumber);
```
Comment (Jojo-Schmitz) [inline importexport/guitarpro/importgtp-gp6.cpp:1228]: How does this work on any plucked string instrument with more than 4 strings that is not a Banjo?

File: importexport/guitarpro/importgtp.cpp [hunk 5]
```diff
@@ -1974,6 +1974,8 @@ bool GuitarPro1::readNote(int string, Note* note)
       // dead note represented as high numbers - fix to zero
       if (fretNumber > 99 || fretNumber == -1)
             fretNumber = 0;
+      if (fretNumber > 0 && string == 4)
+          fretNumber = staff->part()->instrument()->stringData()->adjustBanjo5thFret(fretNumber);
```
Comment (Jojo-Schmitz) [inline importexport/guitarpro/importgtp.cpp:1978]: How does this work on any plucked string instrument with more than 4 strings that is not a Banjo?

File: libmscore/stringdata.cpp [hunk 6]
```diff
@@ -47,6 +47,14 @@ StringData::StringData(int numFrets, QList<instrString>& strings)
             stringTable.append(i);
       }
 
+// called from import (musicxml/guitarpro/...)
+void StringData::set(const StringData& src)
+      {
+          *this = src;
+          if (stringTable.size() == 5)
```
Comment (Jojo-Schmitz) [inline libmscore/stringdata.cpp:54]: What about other plucked string instruments with 5 strings, like some basses?

File: libmscore/stringdata.cpp [hunk 7]
```diff
@@ -67,6 +75,8 @@ void StringData::read(XmlReader& e)
             else
                   e.unknown();
             }
+      if (stringTable.size() == 5)
```
Comment (Jojo-Schmitz) [inline libmscore/stringdata.cpp:78]: What about other plucked string instruments with 5 strings, like some basses?

File: libmscore/stringdata.cpp [hunk 8]
```diff
@@ -388,7 +428,7 @@ int StringData::fret(int pitch, int string, int pitchOffset) const
 //---------------------------------------------------------
 
 void StringData::sortChordNotes(QMap<int, Note *>& sortedNotes, const Chord *chord, int pitchOffset, int* count) const
-{
+      {
```
Comment (Jojo-Schmitz) [inline libmscore/stringdata.cpp:431]: unneeded whitespace change

**Comment by Jojo-Schmitz** [comment]
Also signing the CLA is not optional ;-)

File: importexport/guitarpro/importgtp-gp4.cpp (hunk 2 above)
Comment (laturetab) [reply importexport/guitarpro/importgtp-gp4.cpp:438]: adjustBanjo5thFret() makes sure number of strings == 5 and uses 'startFret' which is only set for 5-string banjos.

File: importexport/guitarpro/importgtp-gp5.cpp (hunk 3 above)
Comment (laturetab) [reply importexport/guitarpro/importgtp-gp5.cpp:1441]: adjustBanjo5thFret() makes sure number of strings == 5 and uses 'startFret' which is only set for 5-string banjos.

File: importexport/guitarpro/importgtp-gp6.cpp (hunk 4 above)
Comment (laturetab) [reply importexport/guitarpro/importgtp-gp6.cpp:1228]: adjustBanjo5thFret() makes sure number of strings == 5 and uses 'startFret' which is only set for 5-string banjos.

File: importexport/guitarpro/importgtp.cpp (hunk 5 above)
Comment (laturetab) [reply importexport/guitarpro/importgtp.cpp:1978]: adjustBanjo5thFret() makes sure number of strings == 5 and uses 'startFret' which is only set for 5-string banjos.

File: libmscore/stringdata.cpp (hunk 6 above)
Comment (laturetab) [reply libmscore/stringdata.cpp:54]: configBanjo5thString() does the needed checks to assure it is a 5 string banjo

File: libmscore/stringdata.cpp (hunk 7 above)
Comment (laturetab) [reply libmscore/stringdata.cpp:78]: configBanjo5thString() does the needed checks to assure it is a 5 string banjo

**Comment by Jojo-Schmitz** [comment]
and just ticking that bix isn't either...

**Comment by Jojo-Schmitz** [comment]
Now those conflict need to get resolved. I guess a `git  pull --rebase musescore 3.x; git push -f` should do (if `musescore` is the name of your upstream repo, could also be `upstream`)

**Comment by laturetab** [comment]
I am completely lost. Maybe I should just start over? I can't resolve the conflicts no matter what I try.

**Comment by Jojo-Schmitz** [comment]
Not with the commands I mentioned?

**Comment by laturetab** [comment]
git pull --rebase  worked
git push -f          didn't work.
git push origin 316896-banjofifthstring worked I think
however is still see the conflicts above

**Comment by laturetab** [comment]
I am going to abort on this and start over. I should have started on 3.x and not 3.6.1. I will also checkout your comments and make better use of diff.

**Pull request closed without merging**